jobs/mongodb_migration/src/mongodb_migration/migrations/_20230216141000_queue_split_names_from_streaming.py

```diff
@@ -18,29 +18,33 @@
 class MigrationQueueUpdateSplitNames(Migration):
     def up(self) -> None:
         logging.info(
-            f"Rename unicity_id field from Job[{split_names}][<dataset>][<config>][<split>] to"
-            f" Job[{split_names_from_streaming}][<dataset>][<config>][<split>]"
+            f"Rename unicity_id field from Job[{split_names}][<dataset>][<config>][None] to"
+            f" Job[{split_names_from_streaming}][<dataset>][<config>][None] and change type from {split_names} to"
+            f" {split_names_from_streaming}"
         )
 
-        for job in Job.objects(type=split_names):
-            job.update(unicity_id=f"Job[{split_names_from_streaming}][{job.dataset}][{job.config}][{job.split}]")
-
-        logging.info(f"Rename type field from {split_names} to {split_names_from_streaming}")
         db = get_db("queue")
-        db["jobsBlue"].update_many({"type": split_names}, {"$set": {"type": split_names_from_streaming}})
+        for job in db["jobsBlue"].find({"type": split_names}):
```

Comment:
I didn't see it first, but I'm afraid the loop will take forever (one query per entry, and there are currently 9.971.923 entries in the collection!) and it will make the migration timeout.

I think that we must try to do a update_many query that updates the unicity_id.

If it's not possible (I don't know much about MongoDB queries), we should delete part of the historical jobs (#818) before running the migration.

Reply:
> I think that we must try to do a update_many query that updates the unicity_id.

Yes, I tried using the $set operator for update the unicity_id with replace and concat (based on dataset/config value) operation but they look to be not working on pymongo, I looked for documentation but I was not able to find a specific one. For mongo the indeed exist doc to update based on functions like $replace $concat but it did'nt worked with pymongo, any ideas? 
I tried this 
` db["jobsBlue"].update_many(
            {"type": "/split-names"},
            {
                "$set": {
                    "unicity_id": {
                        "$replaceOne": {
                            "input": "$unicity_id",
                            "find": "Job[/split-names]",
                            "replacement": "Job[/split-names-from-streaming]",
                        }
                    }
                }
            },
        )` 
But db records were updated with a json in the unicity_id field :/

Reply:
> I think that we should also add one test for each migration script, using a dump of 100 random entries from prod. WDYT? It might be a new issue/PR

Yes for sure we need to test the migrations simulating existing old data

Reply:
cc @coyotte508 for https://github.com/huggingface/datasets-server/pull/837#discussion_r1116903403 maybe

Reply:
Thanks! @coyotte508 it worked

Reply:
> If it's not possible (I don't know much about MongoDB queries), we should delete part of the historical jobs (https://github.com/huggingface/datasets-server/issues/818) before running the migration.

Fixed by https://github.com/huggingface/datasets-server/pull/851. But as the index will be created after the migration job, it will not help here. Maybe we can add it manually to the index in production?

Reply:
I created the index manually in the prod database. Reduced from 10,000,000 entries to 200,000 (last seven days)